homeassistant/components/device_automation/__init__.py

```diff
@@ -42,48 +52,53 @@
     hass.components.websocket_api.async_register_command(
         websocket_device_automation_list_triggers
     )
+    hass.components.websocket_api.async_register_command(
+        websocket_device_automation_get_trigger_capabilities
+    )
     return True
 
 
-async def async_get_device_automation_platform(hass, config, automation_type):
+async def _async_get_device_automation_platform(hass, domain, automation_type):
     """Load device automation platform for integration.
 
     Throws InvalidDeviceAutomationConfig if the integration is not found or does not support device automation.
     """
-    platform_name, _ = TYPES[automation_type]
+    platform_name = TYPES[automation_type][0]
     try:
-        integration = await async_get_integration(hass, config[CONF_DOMAIN])
+        integration = await async_get_integration(hass, domain)
         platform = integration.get_platform(platform_name)
     except IntegrationNotFound:
-        raise InvalidDeviceAutomationConfig(
-            f"Integration '{config[CONF_DOMAIN]}' not found"
-        )
+        raise InvalidDeviceAutomationConfig(f"Integration '{domain}' not found")
     except ImportError:
         raise InvalidDeviceAutomationConfig(
-            f"Integration '{config[CONF_DOMAIN]}' does not support device automation {automation_type}s"
+            f"Integration '{domain}' does not support device automation {automation_type}s"
         )
 
     return platform
 
 
+async def async_get_device_automation_platform(hass, config, automation_type):
```

Comment:
Why does this function exist if it just forwards the call ?

Reply:
I do think that `domain` is a better parameter.

Reply:
This is called from automation platform and condition + script helpers. For example:
https://github.com/home-assistant/home-assistant/blob/bc2755f4ca6c4a6658ac5ec2f5cd871343f18d36/homeassistant/helpers/script.py#L91-93
I thought it was cleaner if they don't need to know what data is needed to load the device automation platform, but maybe it doesn't matter?

Reply:
The knowledge of what those objects contain is already quite spread out between device_automation integration and the script and conditions helpers and the device_automation trigger. I guess it doesn't _really_ matter.

Reply:
Alright, dropped the additional function.